cmake/FindWayland.cmake

```diff
@@ -28,38 +25,35 @@
 # limitations under the License.
 #=============================================================================
 
-set(REQUIRED_WAYLAND_CLIENT_VERSION 1.3.0)
 include(FindPkgConfig)
 
 if(PKG_CONFIG_FOUND)
-  pkg_check_modules(WAYLAND wayland-client)
+    pkg_check_modules(WAYLAND_SCANNER_PKG wayland-scanner)    
+    pkg_check_modules(WAYLAND_CLIENT_PKG wayland-client)
+    pkg_check_modules(XKBCOMMON_PC_PKG xkbcommon)
 endif()
 
-find_path(WAYLAND_INCLUDE_DIR NAMES wayland-client.h
-          PATHS ${WAYLAND_INCLUDE_DIRS}
-          DOC "The Wayland include directory"
+find_program(WAYLAND_SCANNER wayland-scanner
+    HINTS "${WAYLAND_SCANNER_PKG_PREFIX}/bin"
 )
 
-find_library(WAYLAND_LIBRARY NAMES wayland-client
-          PATHS ${WAYLAND_LIBRARY_DIRS}
-          DOC "The Wayland client library"
+find_path(WAYLAND_INCLUDE_DIR wayland-client.h
+    HINTS ${WAYLAND_CLIENT_PKG_INCLUDE_DIRS}
 )
 
-include(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(WAYLAND DEFAULT_MSG WAYLAND_LIBRARY WAYLAND_INCLUDE_DIR)
+find_library(WAYLAND_LIBS 
+    NAMES "wayland-client"
+    HINTS "${WAYLAND_CLIENT_PKG_LIBRARIES}"
```

Comment:
@hardening According to cmake documentation this variable only contains the name of the library suitable to use with ```-l <libname>```, you should use ```WAYLAND_CLIENT_PKG_LIBRARY_DIRS``` instead